src/app/pages/questions/services/audio-record.service.ts

```diff
@@ -1,110 +1,75 @@
 import { Injectable } from '@angular/core'
+import { Device } from '@ionic-native/device/ngx'
 import { File } from '@ionic-native/file/ngx'
-// NOTE: File path to opensmile.js; Adding opensmile plugin
-import * as opensmile from 'cordova-plugin-opensmile/www/opensmile'
 
-declare var cordova: any
-declare var window: any
+declare var Media: any // stops errors w/ cordova-plugin-media-with-compression types
+
 @Injectable()
 export class AudioRecordService {
-  filename: string = 'audio-opensmile.bin'
-
-  audioRecordStatus: boolean = false
-  recording: boolean = false
-  recordingTime: number = 45000
+  isRecording: boolean = false
+  audio
+  fileName = 'audio.m4a'
+  recordingTimeout
 
-  constructor(private file: File) {
+  constructor(private file: File, private device: Device) {
     // NOTE: Kill recording on load
     this.stopAudioRecording()
   }
 
-  setAudioRecordStatus(status) {
-    this.audioRecordStatus = status
-  }
-  getAudioRecordStatus() {
-    return this.audioRecordStatus
-  }
+  startAudioRecording(length): Promise<any> {
+    return new Promise((resolve, reject) => {
+      if (!this.isRecording) this.isRecording = true
+      else reject()
 
-  startAudioRecording(configFile) {
-    this.recording = this.getAudioRecordStatus()
-
-    if (this.recording === false) {
-      this.recording = true
-      this.setRecordTimer()
-      this.setAudioRecordStatus(this.recording)
-      opensmile.start(this.filename, configFile, this.success, this.failure)
-    } else if (this.recording === true) {
-      this.recording = false
-      this.setAudioRecordStatus(this.recording)
-      opensmile.stop('Stop', this.success, this.failure)
-      this.readAudioFile(this.filename)
-    }
+      this.audio = new Media(this.getFilePath(), this.success, this.failure)
+      const options = { SampleRate: 16000, NumberOfChannels: 1 }
```

Comment:
Make this configurable for later changes.

Reply:
I've made the changes.